aggregator/entry.go

```diff
@@ -112,6 +112,53 @@ func (e *Entry) AddMetricWithPoliciesList(
 	mu unaggregated.MetricUnion,
 	pl policy.PoliciesList,
 ) error {
+	switch mu.Type {
+	case unaggregated.BatchTimerType:
+		err := e.writeBatchTimerWithPoliciesList(mu, pl)
+		if mu.BatchTimerVal != nil && mu.TimerValPool != nil {
+			mu.TimerValPool.Put(mu.BatchTimerVal)
+		}
+		return err
+	default:
+		return e.addMetricWithPoliciesList(mu, pl)
+	}
+}
+
+func (e *Entry) writeBatchTimerWithPoliciesList(
+	mu unaggregated.MetricUnion,
+	pl policy.PoliciesList,
+) error {
+	// If there is no limit on the maximum batch size per write, write
+	// all timers at once.
+	maxTimerBatchSizePerWrite := e.opts.MaxTimerBatchSizePerWrite()
```

Comment:
This is primarily to reduce the locking overhead observed when list needs to flush and acquire a write time lock shared by every entry. For a large batch size, it holds the lock for extended period of time and the list cannot flush before that lock is released. By splitting the batches into smaller ones, it gives the flushing list more chances to get the time lock, therefore reducing the average time to acquire the time lock.